tests/libvmi/storage.go

```diff
@@ -28,8 +30,26 @@ import (
 func WithContainerImage(name string) Option {
 	return func(vmi *kvirtv1.VirtualMachineInstance) {
 		diskName := "disk0"
-		vmi.Spec.Domain.Devices.Disks = append(vmi.Spec.Domain.Devices.Disks, newDisk(diskName, v1.DiskBusVirtio))
-		vmi.Spec.Volumes = append(vmi.Spec.Volumes, newContainerVolume(diskName, name))
+		replaceDisk(vmi, newDisk(diskName, v1.DiskBusVirtio))
+		replaceVolume(vmi, newContainerVolume(diskName, name))
+	}
+}
+
+// WithPersistentVolumeClaim specifies the name of the PersistentVolumeClaim to be used.
+func WithPersistentVolumeClaim(name string) Option {
+	return func(vmi *kvirtv1.VirtualMachineInstance) {
+		diskName := "disk0"
```

Comment:
I don't understand why we went with static names. It is not flexible and the usage of our options is very limited by this.

Reply:
It was like that for container images, maybe the assumption was it will always be a single disk? No idea. 
My first idea was to make a small PR to add PVC and DV. Without any refactoring. So I did not change it.

It would require adding name to every call site `WithPersistentVolumeClaim(diskName, pvcName)`. But I think that will be easy.